crates/lsp/src/lsp.rs

```diff
@@ -164,6 +164,24 @@ struct Error {
     message: String,
 }
 
+#[derive(Debug)]
+pub enum ServerStatus {}
+
+#[derive(Debug, PartialEq, Deserialize, Serialize, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct ServerStatusParams {
+    pub health: String, // "ok" | "warning" | "error"
```

Comment:
instead of making this a string, would an enum be more ideal?

Reply:
In that case, the enum should have an `Other(String)` variant since there could be anything in that field due to this notification being non-finalized.

I would be also really interested to know whether there's some description of this command anywhere, have you looked it up?

Reply:
Noted! Yupp! I referred to this: https://github.com/rust-lang/rust-analyzer/blob/master/docs/dev/lsp-extensions.md